### ISSUE TITLE
Need support for tests that target ARM

### ISSUE THREAD

**Issue by aneksteind** [immunant/c2rust#410]
Tests such as `tests/asm.arm` and `tests/asm.aarch64` are unable to execute in the test pipeline because their inline assembly is meant to target ARM.

**Comment by kkysen** [comment]
Can we run these tests using qemu?  And then cross-compile maybe using `zig cc` (through something like [`cargo zigbuild`](https://crates.io/crates/cargo-zigbuild))?

**Comment by aneksteind** [comment]
> Can we run these tests using qemu?

Yes, we discussed this a while ago; the work just needs to be done. One alternative is to add an ARM image/container to the pipeline and test just the relevant tests, which would at least remove the need to cross compile. I don't have a preference which one to choose, probably whichever is easiest.

**Comment by kkysen** [comment]
If we can easily test locally, too, that seems preferable.
